hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/marker/WriteMarkers.java

```diff
@@ -209,6 +209,15 @@ protected StoragePath getMarkerPath(String partitionPath, String fileName, IOTyp
    */
   public abstract Set<String> createdAndMergedDataPaths(HoodieEngineContext context, int parallelism) throws IOException;
 
+  /**
+   * Fetches markers for log files w/ Append IOType. Used only for table version 6.
+   * @param context {@code HoodieEngineContext} instance.
+   * @param parallelism parallelism for reading the marker files in the directory.
+   * @return all the log file paths of write IO type "APPEND"
+   * @throws IOException
+   */
+  public abstract Set<String> getAppendedLogPaths(HoodieEngineContext context, int parallelism) throws IOException;
```

Comment:
This should not exposed here, only to be used by table version 6.

Reply:
Fixed by introducing `AppendMarkerHandler` interface to handle `APPEND` markers for table version 6 only, so that in the future we can easily remove the interface and new implementation classes for V1 only.